File: solutions/c/rna-transcription/4/rna_transcription.c
#include "rna_transcription.h"
#include <stdlib.h>

char *to_rna(const char *dna){
    char* rna = malloc(100*sizeof(char));
    int i=0;
    for (i = 0; dna[i]; i++) {
        switch (dna[i]) {
            case 'G': rna[i] = 'C'; break;
            case 'C': rna[i] = 'G'; break;
            case 'T': rna[i] = 'A'; break;
            case 'A': rna[i] = 'U'; break;
        }
    }
    rna[i]='\0';
    return rna;
}

File: solutions/go/rna-transcription/1/rna_transcription.go
// Package strand transcripts a DNA strand into its RNA complement
package strand

// ToRNA takes a DNA strand as an argument a returns its RNA complement
func ToRNA(dna string) string {
	rna := ""
	for _, dnaC := range(dna) {
		switch dnaC {
		case 'G':
			rna = rna + "C"
		case 'C':
			rna = rna + "G"
		case 'T':
			rna = rna + "A"
		case 'A':
			rna = rna + "U"
		}
	}
	return rna
}
